shawmut/weather.py

```diff
@@ -0,0 +1,43 @@
+from datetime import datetime
+from requests_forecast import Forecast
+from shawmut.settings import conf
+
+class ShawmutWeather(object):
+    MAX_RETRIES = 500 # We are allowed 1000 API calls per today so this is pretty conservative
+
+    def __init__(self):
+        self.refresh_todays_data()
+
+    def get_todays_data(self):
+        if datetime.now() > self.todays_datetime():
+            self.daily_retries = 0
+            self.refresh_todays_data()
+
+    def refresh_todays_data(self):
+        try:
+            forecast = Forecast(apikey=conf.forecast_io_key, latitude=conf.latitude, longitude=conf.longitude)
+            self.todays_data = forecast.get_daily()['data'][0]
+        except Exception as e:
+            if self.retries < MAX_RETRIES:
```

Comment:
This should also raise if we have never successfully retrieved `todays_data` (if `todays_data` is `None`?).

Another option: just make this recursive and keep calling `refresh_todays_data` until it succeeds or goes over `MAX_RETRIES`